abce/trade.pyx

```diff
@@ -544,8 +543,7 @@ class Trade:
                 raise NotEnoughGoods(self.name, offer.good, quantity - available)
             if quantity > available:
                 quantity = available
-            self._haves[offer.good] -= quantity
-            self._haves['money'] += quantity * offer.price
+            self._haves.transform(offer.good, offer.price, 'money', quantity)
```

Comment:
I suggest we implement `Inventory.__add__` and `__subtract__`. Then we don't have to use destroy transform and create where they don't actually make sense, because a good is handed over rather then destroyed and created.